File: src/ch03/solution3_2.py
from typing import List

from rich import print


def count_num(nums: List[int], target: int) -> int:
    counter: int = 0
    for num in nums:
        if num == target:
            counter += 1
    
    return counter


if __name__ == '__main__':
    nums: List[int] = [3, 4, 1, 2, 5, 1]
    target: int = 1
    result: int = count_num(nums=nums, target=target)
    print(result)

File: src/ch03/solution3_2.rs
fn count_num(nums: Vec<i32>, target: i32) -> i32 {
    let mut counter: i32 = 0;
    for num in nums {
        if num == target {
            counter += 1;
        }
    }
    counter
}

fn main() {
    let nums: Vec<i32> = vec![3, 4, 1, 2, 5, 1];
    let target: i32 = 1;
    let result: i32 = count_num(nums, target);

    println!("{}", result);
}